src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs

```diff
@@ -301,6 +305,21 @@ private static bool IsDistroAndVersionOrHigher(Predicate<string> distroPredicate
             return false;
         }
 
+        private static bool IsQemu()
+        {
+            if (IsLinux)
+            {
+                foreach (DictionaryEntry entry in Environment.GetEnvironmentVariables())
```

Comment:
This should just call `Environment.GetEnvironmentVariable("DOTNET_RUNNING_UNDER_QEMU") != null`.